raft/src/test/java/org/apache/kafka/raft/NoOpStateMachine.java

```diff
@@ -19,17 +19,26 @@
 import org.apache.kafka.common.record.RecordBatch;
 import org.apache.kafka.common.record.Records;
 
+import java.util.Collections;
+import java.util.Optional;
+
 public class NoOpStateMachine implements DistributedStateMachine {
     private OffsetAndEpoch position = new OffsetAndEpoch(0, 0);
 
+    private Optional<EpochState> state = Optional.empty();
+
     @Override
     public void becomeLeader(int epoch) {
-
+        this.state = Optional.of(new LeaderState(0, epoch, 0L, Collections.emptySet()));
```

Comment:
I don't think we want state machines using internal classes like `LeaderState` and `FollowerState`. Could we just create an internal class here or use private fields to track this?